shows_app/src/app/components/feature/shows/ReviewForm/ReviewForm.tsx

```diff
@@ -9,74 +9,63 @@ import {
   Stack,
   Text
 } from "@chakra-ui/react";
-
 import { StarRating } from "../../reviews/StarRating/StarRating";
-import { IReviewContent } from "../../../../typings/reviews";
-
+import { IReview, IReviewContent } from "../../../../typings/reviews";
+import { useForm } from "react-hook-form";
 
 export interface IReviewFormProps {
-  addShowReview: (review: IReviewContent) => void;
+  addShowReview: (review: IReview) => void;
 }
 
 export const ReviewForm = ({ addShowReview }: IReviewFormProps) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors, isSubmitting }
+  } = useForm<IReview>();
   const [rating, setRating] = useState<number>(0);
-  const [comment, setComment] = useState<string>("");
- 
 
-  const setRatingchange = (newRating: number) => {
-    setRating(newRating);
-  };
-
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    addShowReview({ rating, comment });
+  const onSubmit = (data: IReview) => {
+    addShowReview({ ...data, rating });
```

Comment:
Similar to what I wrote for deleting reviews, we can have the swrMutation that does this in the reviewItem component. The only thing we have to take care of is to tell SWR that it should refresh what it is keeping in the `reviews` cache because we just deleted one. We can do that with `mutate`.